scripts/dataset_processing/ljspeech/extract_ljspeech_phonemes_and_durs.sh

```diff
@@ -0,0 +1,141 @@
+# Also requires that you have manifests set up pointing to the appropriate
+# train/val/test wav files. Make sure you have run the
+# `create_manifests_and_textfiles.py` script.
+#
+# This script will create:
+# - <LJSPEECH_BASE>/mappings.json: Contains word->phone and phone->idx mappings
+# - <LJSPEECH_BASE>/phoneme_durations/LJ*.npz: Numpy files for each wavfile
+#     with fields 'tokens' (token indices) and 'durs' (durations per token)
+#
+# It will also create intermediate files:
+# - <LJSPEECH_BASE>/wavs/LJ*.txt: Transcripts per wavfile (forA MFA alignment)
+# - <LJSPEECH_BASE>/alignments/wavs/LJ*.TextGrid: MFA output
+# - <LJSPEECH_BASE>/ljspeech_dict.txt: MFA-generated word->phone mapping
+#
+# Note: This script will create a conda environment to set up MFA, so please
+#       make sure that conda is active before running this script.
+#
+# Note: You may need to install OpenBlas yourself if the MFA script can't find
+#       it. (e.g. sudo apt-get install libopenblas-dev)
+#
+# Example Usage:
+# ./extract_ljspeech_phonemes_and_durs.sh \
+#   --train_manifest=/data/manifests/ljspeech_train.json \
+#   --val_manifest=/data/manifests/ljspeech_val.json \
+#   --test_manifest=/data/manifests/ljspeech_test.json \
+#   /data/LJSpeech-1.1
+
+ENV_NAME='aligner'
+
+SAMPLE_RATE=22050
+WINDOW_STRIDE=256
+
+CMUDICT_URL='https://raw.githubusercontent.com/cmusphinx/cmudict/master/cmudict.dict'
+SPLITS_BASE_URL='https://raw.githubusercontent.com/NVIDIA/tacotron2/master/filelists/'
+
+# Usage info
+show_help() {
+cat << EOF
+Usage: $(basename "$0") [-h] \
+          [--skip_env_setup] \
+          [--g2p_dict=<G2P_DICT_PATH>] \
+          <LJSPEECH_BASE>
+Extracts phonemes and their respective durations for the LJSpeech dataset using
+the Montreal Forced Aligner (MFA).
+
+    -h                Help message
+    --skip_env_setup  (Optional) Skips setting up the MFA conda environment
+                      "aligner".  Use only if you already have this set up.
+    --g2p_dict        (Optional) Path to the grapheme to phoneme dictionary
+                      text file, if already generated. If set, skips the G2P
+                      step.
+EOF
+}
+
+SKIP_ENV_SETUP=false
+G2P_DICT=''
+
+TRAIN_MANIFEST=''
+VAL_MANIFEST=''
+TEST_MANIFEST=''
+
+while :; do
+  case $1 in
+    -h|-\?|--help)
+      show_help
+      exit
+      ;;
+    --g2p_dict=?*)
+      G2P_DICT=${1#*=}
+      ;;
+    --skip_env_setup)
+      SKIP_ENV_SETUP=true
+      ;;
+    *)
+      break
+  esac
+  shift
+done
+
+if [[ -z $1 ]]; then
+  echo "Must specify a LJSpeech base directory."
+  exit 1
+fi
+LJSPEECH_BASE=$1
+
+# Check for conda
+read -r -d '' CONDA_MESSAGE << EOM
+This script requires either Anaconda or Miniconda to be active, as it installs the Montreal Forced Aligner via a conda environment.
+See their documentation (https://montreal-forced-aligner.readthedocs.io/en/latest/installation.html) for more details.
+EOM
+conda -h > /dev/null || echo $CONDA_MESSAGE
```

Comment:
Let's exit here if conda is not found

Reply:
Line 97 should take care of that by catching the error and printing out that message, no?

Reply:
It does not. On my machine, it just prints the error and then continues on to the rest of the script

Reply:
Hm, alright. Changing this to a conditional that exits explicitly.